project/CONTRIBUTING.md

```diff
@@ -232,8 +232,62 @@ interface BarInterface
 }
 ```
 
-In some cases, you will have the possibility to warn the user that things will change,
-and recommend a new way of doing things. You can do so by triggering the dedicated kind of error, like this:
+##### Deprecation
+
+In some cases, you might have some code part being useless.
```

Comment:
What do you mean? Why shouldn't we remove it directly if it is useless. To me, useless = cannot be reached.

Reply:
Example: When we moved `Admin` to `AbstractAdmin`, the `Admin` class is now useless. But can be still used by the end user.

Did you got my point? How would you explain it differently?

Reply:
I'll trust you on this one. :wink: